actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -298,7 +299,7 @@ def raw_post
     # variable is already set, wrap it in a StringIO.
     def body
       if raw_post = get_header("RAW_POST_DATA")
-        raw_post.force_encoding(Encoding::BINARY)
+        raw_post = raw_post.dup.force_encoding(Encoding::BINARY)
```

Comment:
I'm torn on this one.. does the change belong here, or is it the caller's responsibility to provide an unfrozen value?